x/dispensation/types/types.pb.go

```diff
@@ -0,0 +1,1673 @@
+// Code generated by protoc-gen-gogo. DO NOT EDIT.
+// source: sifnode/dispensation/v1/types.proto
+
+package types
+
+import (
+	fmt "fmt"
+	github_com_cosmos_cosmos_sdk_types "github.com/cosmos/cosmos-sdk/types"
+	types "github.com/cosmos/cosmos-sdk/types"
+	_ "github.com/gogo/protobuf/gogoproto"
+	proto "github.com/gogo/protobuf/proto"
+	io "io"
+	math "math"
+	math_bits "math/bits"
+)
+
+// Reference imports to suppress errors if they are not otherwise used.
+var _ = proto.Marshal
+var _ = fmt.Errorf
+var _ = math.Inf
+
+// This is a compile-time assertion to ensure that this generated file
+// is compatible with the proto package it is being compiled against.
+// A compilation error at this line likely means your copy of the
+// proto package needs to be updated.
+const _ = proto.GoGoProtoPackageIsVersion3 // please upgrade the proto package
+
+// Distribution type enum
+type DistributionType int32
+
+const (
+	// Unspecified distribution type
+	DistributionType_DISTRIBUTION_TYPE_UNSPECIFIED DistributionType = 0
+	// Airdrop distribution type
+	DistributionType_DISTRIBUTION_TYPE_AIRDROP DistributionType = 1
+)
+
+var DistributionType_name = map[int32]string{
+	0: "DISTRIBUTION_TYPE_UNSPECIFIED",
+	1: "DISTRIBUTION_TYPE_AIRDROP",
+}
+
+var DistributionType_value = map[string]int32{
+	"DISTRIBUTION_TYPE_UNSPECIFIED": 0,
+	"DISTRIBUTION_TYPE_AIRDROP":     1,
+}
+
+func (x DistributionType) String() string {
+	return proto.EnumName(DistributionType_name, int32(x))
+}
+
+func (DistributionType) EnumDescriptor() ([]byte, []int) {
+	return fileDescriptor_bfdf912039cd8799, []int{0}
+}
+
+// Claim status enum
+type ClaimStatus int32
+
+const (
+	// Pending claim status
+	ClaimStatus_CLAIM_STATUS_PENDING ClaimStatus = 0
+	// Completed claim status
+	ClaimStatus_CLAIM_STATUS_COMPLETED ClaimStatus = 1
+)
+
+var ClaimStatus_name = map[int32]string{
+	0: "CLAIM_STATUS_PENDING",
+	1: "CLAIM_STATUS_COMPLETED",
+}
+
+var ClaimStatus_value = map[string]int32{
+	"CLAIM_STATUS_PENDING":   0,
+	"CLAIM_STATUS_COMPLETED": 1,
+}
+
+func (x ClaimStatus) String() string {
+	return proto.EnumName(ClaimStatus_name, int32(x))
+}
+
+func (ClaimStatus) EnumDescriptor() ([]byte, []int) {
+	return fileDescriptor_bfdf912039cd8799, []int{1}
+}
+
+type GenesisState struct {
```

Comment:
Why do we need this in the genesisstate for dispesation ?

Reply:
@Jedi2002 Can you take on the genesis export / import implementation on this PR?